tests/router_test.js

```diff
@@ -4595,6 +4596,84 @@ test("A failed handler's setup shouldn't prevent future transitions", function(a
     assert.equal(barModelCount, 1, 'Bar model should be called once');
   });
 
+  test('Calling replaceWith after initial replace in validation hook with multiple redirects should use replaceUrl', function(assert) {
+    // assert.expect(8);
+
+    map(assert, function(match) {
+      match('/foo').to('foo');
+      match('/bar').to('bar');
+      match('/baz').to('baz');
+      match('/qux').to('qux');
+    });
+
+    var fooModelCount = 0,
+      barModelCount = 0,
+      bazModelCount = 0,
+      history = [];
+
+    router.updateURL = function(updateUrl) {
+      url = updateUrl;
+      history.push(url);
+    };
+
+    router.replaceURL = function(replaceURL) {
+      url = replaceURL;
+      if (history.length > 0) {
+        history[history.length - 1] = url;
+      } else {
+        history.push(url);
```

Comment:
Is this actually correct? My understanding of the current solution to "redirects during initial transition" is that we _will not_ call replaceURL prior to updateURL. If that is still correct then this conditional should not be needed...

Reply:
TBH, I wasn't thinking that carefully about that bit. I can remove it and see how it goes.

Reply:
Yep, sounds good to me.